storagenode/console/consoleserver/server.go

```diff
@@ -69,176 +93,290 @@ func NewServer(logger *zap.Logger, config Config, service *console.Service, list
-			DiskSpace     console.DiskSpaceInfo `json:"diskSpace"`
-			WalletAddress string                `json:"walletAddress"`
-			VersionInfo   version.Info          `json:"versionInfo"`
-			IsLastVersion bool                  `json:"isLastVersion"`
-			Uptime        time.Duration         `json:"uptime"`
-			NodeID        string                `json:"nodeId"`
-			Satellites    storj.NodeIDList      `json:"satellites"`
-		} `json:"data"`
-		Error string `json:"error,omitempty"`
-	}
+	writer.Header().Set(contentType, applicationJSON)
+
+	var response = DashboardResponse{}
 
 	defer func() {
-		err := json.NewEncoder(w).Encode(&response)
+		err := json.NewEncoder(writer).Encode(&response)
 		if err != nil {
-			s.log.Error(err.Error())
+			server.log.Error(err.Error())
 		}
 	}()
 
-	if req.Method != http.MethodGet {
-		w.WriteHeader(http.StatusNotFound)
+	if request.Method != http.MethodGet {
+		writer.WriteHeader(http.StatusNotFound)
 		return
 	}
 
-	satelliteIDParam := req.URL.Query().Get("satelliteId")
-	satelliteID, err := s.parseSatelliteIDParam(satelliteIDParam)
+	satelliteIDParam := request.URL.Query().Get("satelliteId")
+	satelliteID, err := server.parseSatelliteIDParam(satelliteIDParam)
 	if err != nil {
-		s.log.Error("satellite id is not valid", zap.Error(err))
+		server.log.Error("satellite id is not valid", zap.Error(err))
 		response.Error = "satellite id is not valid"
-		w.WriteHeader(http.StatusBadRequest)
+		writer.WriteHeader(http.StatusBadRequest)
 		return
 	}
 
-	satellites, err := s.service.GetSatellites(ctx)
+	data, err := server.getDashboardData(ctx, satelliteID)
 	if err != nil {
-		s.log.Error("can not get satellites list", zap.Error(err))
-		response.Error = "can not get satellites list"
-		w.WriteHeader(http.StatusInternalServerError)
-		return
+		server.log.Error("can not get dashboard data", zap.Error(err))
+		response.Error = err.Error()
+	}
+
+	response.Data = data
+
+	writer.WriteHeader(http.StatusOK)
+}
+
+// appHandler is web app http handler function
+func (server *Server) liveReloadHandler(writer http.ResponseWriter, request *http.Request) {
+	ctx := request.Context()
+	defer mon.Task()(&ctx)(nil)
+
+	var response = &DashboardResponse{}
+
+	conn, err := server.upgrader.Upgrade(writer, request, nil)
+	if err != nil {
+		server.log.Error("\nwebsocket error: ", zap.Error(err))
+	}
+
+	var satelliteID *storj.NodeID
+
+	go func(conn *websocket.Conn) {
+		for {
+			_, satelliteIDParam, err := conn.ReadMessage()
+			if err != nil {
+				if err = conn.Close(); err != nil {
+					server.log.Error("can not close websocket connection", zap.Error(err))
+				}
+			}
+
+			satelliteID, err = server.parseSatelliteIDParam(string(satelliteIDParam))
+			if err != nil {
+				server.log.Error("\nsatellite id is not valid", zap.Error(err))
+				response.Error = "satellite id is not valid"
+			}
+		}
+	}(conn)
+
+	go func(conn *websocket.Conn) {
+		ch := time.Tick(5 * time.Second)
+
+		for range ch {
```

Comment:
should we handle `ctx` cancellation here and use `select` statement instead of `range`?

Reply:
Since we query data from `satellite` 5 seconds might be too much